JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/slurm/job/JobSubmission.java

```diff
@@ -439,6 +443,17 @@ public void monitorJobs() throws SlurmJobException{
 				session = jsch.getSession(slurmJobProperty.getHpcServerLoginUserName(), getHpcAddress(), 22);
 				String pwd = slurmJobProperty.getHpcServerLoginUserPassword();
 				session.setPassword(pwd);
+
+				try {
+					Connector con = new PageantConnector();
+					IdentityRepository irepo = new RemoteIdentityRepository(con);
+					jsch.setIdentityRepository(irepo);
+					session.setConfig("PreferredAuthentications", "publickey,keyboard-interactive,password");
```

Comment:
Hardcoded values need to be replaced with variables. In Java, a good practice is to store them in properties file.

Reply:
Whilst I agree that hardcoding settings is bad in general I'm not sure making this setting available to users would be beneficial. The only advantage I could see for doing so would be to allow the user to force authentication using the password even when a valid public key is loading into Pageant. I will add some comments to make this clearer in the code.

Reply:
I haven't mentioned exposing the functionality to users. Just not hardcoding strings.

Reply:
Sorry, when I say users, I'm thinking of the developers using this code. Hopefully the code comments I've suggested in GitHub comment below make it clearer why those strings are hardcoded and not specified in a properties file.

Reply:
If you insist of not using properties, please at least use final class member variables for them (aka constants). You'll be able to reuse them in the code. Otherwise you're repeating the hardcoded strings.

Reply:
Whilst I agree that in general string constants should be made static final member variables I think that in this instance using the raw strings is clearer and will make it easier to add new authentication types in the future. These strings are not intended for use outside of this method and are very context specific. I also believe that making them variables will lead to potential confusion as the names of those variables would be intrinsically tied to their content and if someone changed the content (for example by adding a new authentication option) but did not change the name of the variable it might become misleading.
There would be a minor benefit to creating constants for each of the options in that it would reduce the probability of mistyping the strings but I think the benefits would be greatly outweighed by having to write code like:
`PUBLICKEY + ',' + KEYBOARD_INTERACTIVE + ',' + PASSWORD`
or
`Stream.of(PUBLICKEY, KEYBOARD_INTERACTIVE, PASSWORD).collect(Collectors.joining(",")`
especially considering the small likelihood of these particular strings being reused.
Finally, in all of the online examples I have seen that discuss this setting they all write the string argument in the same form as I have and so I feel that keeping it like this will make it easier for any future developers to compare the current settings to those in the examples they see.
I accept that my code is not always correct, let alone perfect, and appreciate that you are only trying to improve the general standard of the World Avatar code, which I fully support and am more than happy to work with you on, but blindly applying simple rules is not the way to improve code quality.

Reply:
Using constants makes it easier to modify code. DRY also affects compilation in Java. Stay consistent and use finals please.